### PR TITLE
feat: Large Rubocop update

### PR DESCRIPTION
**Pull request by zverok**
* Update versions to the latest gems
* Reiterate on what's good and what's not, update settings

The plan is:
* https://github.com/NetsoftHoldings/hubstaff-server/pull/5421 is reviewed and merged (new approach to structure TODOs + update styles), depending on the new version
* We make sure it works for the server (after the merge)
* One-by-one other repos are switched to this version and updated in the same fashion (several structured TODO files)

### CODE REVIEW

File: netsoft-rubocop.gemspec [hunk 1]
```diff
@@ -15,18 +15,21 @@ Gem::Specification.new do |spec|
 
   # Specify which files should be added to the gem when it is released.
   # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path(__dir__)) do
+  spec.files         = Dir.chdir(File.expand_path(__dir__)) {
     `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
-  end
+  }
```
Comment (sergeygaychuk) [inline netsoft-rubocop.gemspec:20]: At least say to me why?

Comment (zverok) [reply netsoft-rubocop.gemspec:20]: New rubocop wanted it this way :shrug: (braces for functional blocks)

**Comment by sergeygaychuk** [comment]
Have you seen PR's Build failed? @zverok

**Comment by sergeygaychuk** [comment]
In general I can approve it. For me looks generally good

**Review by urkle — APPROVED** [review]

**Pull request merged**